src/tablemodel.cpp

```diff
@@ -27,69 +27,70 @@
 #include "dlt_protocol.h"
 #include "qdltoptmanager.h"
 
+// this is done to have a basic caching mechanism in the table for the cases when the same dlt
+// message is accessed multiple times in the TableModel::data function several times in a row
+// TODO: replace with some kind of LRU cache to make it explicit
 static long int lastrow = -1; // necessary because object tablemodel can not be changed, so no member variable can be used
-char buffer[DLT_VIEWER_LIST_BUFFER_SIZE];
 
-
-void getmessage( int indexrow, long int filterposindex, unsigned int* decodeflag, QDltMsg* msg, QDltMsg* lastmsg, QDltFile* qfile, bool* success )
+void getMessage(int indexrow, QDltMsg* msg, QDltMsg* lastmsg, QDltFile* qfile, bool* success )
 {
- if ( indexrow == lastrow)
- {
-  *msg = *lastmsg;
- }
- else
- {
-  *success = qfile->getMsg(filterposindex, *msg);
-  *lastmsg = *msg;
-  *decodeflag = 1;
- }
- if ( indexrow == 0)
- {
-  lastrow = 0;
- }
- else
- {
-  lastrow = indexrow;
- }
-}
+    if (indexrow == lastrow)
+    {
+        *msg = *lastmsg;
+    }
+    else
+    {
+        auto filterposindex = qfile->getMsgFilterPos(indexrow);
+        *success = qfile->getMsg(filterposindex, *msg);
+        *lastmsg = *msg;
+    }
 
+    lastrow = indexrow;
+}
 
 TableModel::TableModel(const QString & /*data*/, QObject *parent)
-     : QAbstractTableModel(parent)
- {
-     qfile = NULL;
-     project = NULL;
-     pluginManager = NULL;
-     lastSearchIndex = -1;
-     emptyForceFlag = false;
-     loggingOnlyMode = false;
-     searchhit = -1;
-     lastrow = -1;
- }
-
- TableModel::~TableModel()
- {
-
- }
+    : QAbstractTableModel(parent)
+{
+    qfile = NULL;
+    project = NULL;
+    pluginManager = NULL;
+    lastSearchIndex = -1;
+    emptyForceFlag = false;
+    loggingOnlyMode = false;
+    searchhit = -1;
+    lastrow = -1;
+}
 
+TableModel::~TableModel() = default;
 
- int TableModel::columnCount(const QModelIndex & /*parent*/) const
- {
-     return DLT_VIEWER_COLUMN_COUNT+project->settings->showArguments;
- }
-
+int TableModel::columnCount(const QModelIndex & /*parent*/) const
+{
+    return DLT_VIEWER_COLUMN_COUNT + project->settings->showArguments;
+}
 
- QVariant TableModel::data(const QModelIndex &index, int role) const
+QVariant TableModel::data(const QModelIndex &index, int role) const
  {
-     QByteArray buf;
      static QDltMsg msg;
      static QDltMsg lastmsg;
      static QDltMsg last_decoded_msg;
      static unsigned int decodeflag = 0;
      static bool success = true;
 
-     long int filterposindex = 0;
-
+     auto decodeMessageWithPlugin = [&] {
+         if (QDltSettingsManager::getInstance()->value("startup/pluginsEnabled", true).toBool())
+         {
+             if ( decodeflag == 1 )
+             {
+                 decodeflag = 0;
+                 pluginManager->decodeMsg(msg, !QDltOptManager::getInstance()->issilentMode());
```

Comment:
@alexmucde initially I had a look to this file to understand if I can remove usage of QDltOptManager singleton in here. this is the only place it is used. Do you know how issilentmode command option relates to triggeredByUser argument of the decodeMsg? Can I simply pass `false` here? I know it is not used in the existing plugins, however I am not sure about third-party ones